bin/test-build-on-all-distros

```diff
@@ -41,21 +41,24 @@ fi
 
 WORK_DIR="$(mktemp -dt hhvm.XXXXXXXX)"
 
+if which gtar
+then
+  TAR=gtar # the executable of GNU Tar in Homebrew 
+else
+  TAR=tar
+fi
 (
   set -x
   cd "$SOURCE_DIR"
-  git checkout-index -a -f --prefix="$WORK_DIR/hhvm-$VERSION/"
+  git ls-files --recurse-submodules |
+  egrep --invert-match -e '^(third-party/boost/boost/libs/[^/]+/(doc|test)|third-party/fb-mysql/mysql-5.6/(boost|rocksdb|xtrabackup|rqg|mysql-test)|hphp/test/(slow|quick|zend|zend7)/|"hphp/test/slow/)' |
+  "$TAR" caf "$WORK_DIR"/hhvm-"$VERSION".tar.gz --xform s:^:hhvm-"$VERSION"/: -T-
 )
 
-"$(dirname "$0")/prune-source-tree" "$WORK_DIR/hhvm-$VERSION"
-
-pushd "$WORK_DIR" >/dev/null
 (
   set -x
-  tar czf "hhvm-$VERSION.tar.gz" "hhvm-$VERSION"
-  aws s3 cp "hhvm-$VERSION.tar.gz" s3://hhvm-scratch/
+  aws s3 cp "$WORK_DIR"/hhvm-"$VERSION".tar.gz s3://hhvm-scratch/
```

Comment:
`"${WORK_DIR}/hhvm-${VERSION}.tar.gz"` is more common - making it one string with interpolation, rather than a concatenation of 4 strings